core/jvmMain/src/kotlinx/serialization/internal/Platform.kt

```diff
@@ -29,10 +28,7 @@ internal actual fun <T : Any, E : T?> ArrayList<E>.toNativeArrayImpl(eClass: KCl
 internal actual fun KClass<*>.platformSpecificSerializerNotRegistered(): Nothing = serializerNotRegistered()
 
 internal fun Class<*>.serializerNotRegistered(): Nothing {
-    throw SerializationException(
-        "Serializer for class '${simpleName}' is not found.\n" +
-                "Mark the class as @Serializable or provide the serializer explicitly."
-    )
+    throw SerializationException(this::class.notRegisteredMessage())
```

Comment:
Looks like this is an error: `this::class` is a `Class::class`

Reply:
Good catch